portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -20,89 +20,116 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import java.util.ArrayList;
+import java.util.List;
 import com.google.gson.Gson;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime; 
+import java.lang.Math;
 
-/** Servlet that handles comment data. */
+/** 
+  * Servlet that uploads and retrieves persistent comment data using datastore.
+  */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  /** Represents a single comment */
+  private static class Comment {
+      String text;
+      String name;
+      String time;
```

Comment:
Storing times as a string is problematic because there are many operations that are often done on times, such as sorting, comparing, and addition/subtraction that are done on other data types. 

Java8 has a good date library, or Longs on their own are OK as an alternative though.

Reply:
time is now stored as a long.